src/GraphQL/EnumerableExtensions.cs

```diff
@@ -10,7 +10,8 @@ public static class EnumerableExtensions
     {
         public static IEnumerable<object> Map(this IEnumerable items, Func<object, object> map)
         {
-            return from object item in items select map(item);
+            var mappedItems = from object item in items select map(item);
+            return mappedItems.ToList();
```

Comment:
Forcing enumeration here as this bit was causing some trouble further down the stack in some cases (can't remember the exact steps to reproduce right now, but can try to find those if you want me to).